### DISCUSSION TITLE
Issues with pasta settings and podman network

### DISCUSSION THREAD

**Discussion by matteo-gambarutti** [containers/podman#23697] [Q&A]
I'm trying to implement the Grafana-Prometheus-Node-exporter stack with podman. Grafana and Prometheus are containers whereas node_exporter is not.
I'd like to have both Grafana and Prometheus in the same podman network so I use containers names.

Node_exporter exposes port 9100.

In the prometheus quadlet file, I can specify `Network=pasta:-T,9100-9100` and with this I can refer to node_exporter with localhost because of the port forwarding. However, I also need to join a podman network called `gp.network`.
So I changed that line to `Network=gp.network` and moved the pasta settings to `~/.config/containers/containers.conf.d/0_network.conf` with the following content:

```
[Network]
pasta_options = ["-T","9100-9100"]
```

With these settings I can't access the metrics of node_exporter from Prometheus and even I can't access Prometheus from Grafana using the container name nor using the host IP address.

Is what I'm trying to achieve possible? What am I doing wrong?

**Comment by sbrivio-rh** [comment]
This diagram https://github.com/containers/podman/discussions/22943#discussioncomment-9795883 shows what happens when you create a custom network and connect that with user-mode networking. Once you have that, you can't refer to "the host" using a loopback address anymore: you need to use a local (not loopback!) address that's configured on the host.
